### ISSUE TITLE
Think about and solve for restricting access to tasks

### ISSUE THREAD

**Issue by dspasojevic** [agiledigital-labs/aws-durable-lambda#103]
In some (lots?) of cases, we will want to restrict who can retrieve the status of the task.

**Comment by NoxHarmonium** [comment]
Good point, we currently have a very coarse grained approach to auth. I think the only security we support is a global API key for the API.

I guess this is another thing that will be much easier with https://github.com/agiledigital-labs/aws-durable-lambda/issues/107
